### PR TITLE
[autopaste] Disallow thread author from deleting the message

### PR DESCRIPTION
**Pull request by Soheab**
This PR changes the delete message view interaction check to disallow the thread author from deleting the paste message and instead allow the help mod to do so (only in help threads).

This is tested.

### CODE REVIEW

File: cogs/autopaste.py [hunk 1]
```diff
@@ -64,11 +64,17 @@ async def interaction_check(self, interaction: Interaction) -> bool:
             await self.on_timeout()
             return False
 
-        if (
-            interaction.user.id == self.message_author.id
-            or interaction.channel.permissions_for(interaction.user).manage_messages
-        ):  # type: ignore
-            return True
+        if isinstance(interaction.channel, Thread) and interaction.channel.parent_id == HELP_CHANNEL_ID:
+            if interaction.user.get_role(HELP_MOD_ID):
```
Comment (onerandomusername) [inline cogs/autopaste.py:68]: should apply the manage message check before the thread author as well.

Comment (Soheab) [reply cogs/autopaste.py:73]: reworked the check to always allow users with manage_messages. thanks

**Review by koala9712 — APPROVED** [review]
Makes sense yes

**Review by ooliver1 — APPROVED** [review]

**Comment by TAG-Epic** [comment]
Can this be changed to allow no deletions in help threads at all?

**Comment by Soheab** [comment]
> Can this be changed to allow no deletions in help threads at all?

That would be unfair for users that don't want the ~~annoying~~ message for every codeblock they send in the threads.
I think this should only be for the author since they are usually the ones sending the imported stuff.

**Pull request merged**